### DISCUSSION TITLE
iOS app?

### DISCUSSION THREAD

**Discussion by xplosionmind** [espanso/espanso#1211] [Feature requests and Ideas]
I have seen that [in the long term plans](https://github.com/federico-terzi/espanso/issues/255) for Espanso there is an Android app.   
Is an iOS version of the app being considered, too?

**Comment by federico-terzi** [comment]
Hey @xplosionmind,

Thank you for the feedback! I've been thinking about the iOS port for a while, and although I'm definitely not against it, it's probably the last one that will get implemented. The reason is that there is no "easy" way to support a third-party text expansion mechanism on iOS. For example, some of the commercial text expanders go as far as implementing a custom keyboard with the expansion mechanism built-in, which is not without problems (for example, it doesn't work with external keyboards).

Given that implementing a keyboard that supports most languages will be pretty hard, I prefer to stick to the platforms I know will yield the best results first.

That said, I'd love to use espanso on my iPad, so won't stop thinking about it :)

**Comment by xplosionmind** [comment]
Thank you very much for the follow up!
